MSAL/test/app/mac/MSALAcquireTokenViewController.m

```diff
@@ -0,0 +1,366 @@
+@property (weak) IBOutlet NSSegmentedControl *validateAuthority;
+@property (weak) IBOutlet NSStackView *acquireTokenView;
+@property (weak) IBOutlet WKWebView *webView;
+
+
+@property MSALTestAppSettings *settings;
+@property NSArray *selectedScopes;
+
+@end
+
+@implementation MSALAcquireTokenViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.settings = [MSALTestAppSettings settings];
+    [self populateProfiles];
+    self.selectedScopes = @[defaultScope];
+}
+
+- (void)populateProfiles
+{
+    [self.profiles removeAllItems];
+    [self.profiles setTarget:self];
+    [self.profiles setAction:@selector(selectedProfileChanged:)];
+    [[MSALTestAppSettings profiles] enumerateKeysAndObjectsUsingBlock: ^(id key, id obj, BOOL *stop) {
+        [self.profiles addItemWithTitle:key];
+    }];
+    
+    [self.profiles selectItemWithTitle:[MSALTestAppSettings currentProfileName]];
+    self.clientId.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:clientId];
+    self.redirectUri.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:redirectUri];
+}
+
+- (IBAction)selectedProfileChanged:(id)sender
+{
+    [self.settings setCurrentProfile:[self.profiles indexOfSelectedItem]];
+    self.clientId.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:clientId];
+    self.redirectUri.stringValue = [[MSALTestAppSettings currentProfile] objectForKey:redirectUri];
+}
+
+- (void)prepareForSegue:(NSStoryboardSegue *)segue sender:(id)sender
+{
+    if ([segue.identifier isEqualToString:@"addScopesSegue"])
+    {
+        MSALScopesViewController *scopesController = (MSALScopesViewController *)segue.destinationController;
+        scopesController.delegate = self;
+    }
+}
+
+- (void)setScopes:(NSMutableArray *)scopes
+{
+    if ([scopes count])
+    {
+        NSString *selectedScopes = [scopes componentsJoinedByString:@","];
+        [self.scopesLabel setStringValue:selectedScopes];
+        self.selectedScopes = scopes;
+    }
+    else
+    {
+        [self.scopesLabel setStringValue:defaultScope];
+        self.selectedScopes = @[defaultScope];
+    }
+}
+
+- (void)updateResultView:(MSALResult *)result
+{
+    NSString *resultText = [NSString stringWithFormat:@"{\n\taccessToken = %@\n\texpiresOn = %@\n\ttenantId = %@\n\tuser = %@\n\tscopes = %@\n\tauthority = %@\n}",
+                            [result.accessToken msidTokenHash], result.expiresOn, result.tenantId, result.account, result.scopes, result.authority];
+    
+    [self.resultView setString:resultText];
+    
+    NSLog(@"%@", resultText);
+}
+
+- (void)updateResultViewError:(NSError *)error
+{
+    NSString *resultText = [NSString stringWithFormat:@"%@", error];
+    [self.resultView setString:resultText];
+    NSLog(@"%@", resultText);
+}
+
+- (MSALPromptType)promptType
+{
+    NSString *promptType = [self.promptBehavior labelForSegment:[self.promptBehavior selectedSegment]];
+    
+    if ([promptType isEqualToString:@"Select"])
```

Comment:
nit: another technique is to create an enum with segmented control indexes and do a switch statement based on that enum

Reply:
Keeping this the same as it does not change any functionality.:)